jenkins/jobs/clusterctl_upgrade.pipeline

```diff
@@ -0,0 +1,111 @@
+import java.text.SimpleDateFormat
+import java.security.SecureRandom
+
+ci_git_credential_id = "metal3-jenkins-github-token"
+
+// 3 hours
+int TIMEOUT = 10800
+
+KEEP_TEST_ENV = (env.KEEP_TEST_ENV)
+
+script {
+    if ("${PROJECT_REPO_ORG}" == 'metal3-io' && "${PROJECT_REPO_NAME}" == 'project-infra') {
+        echo "Checkout ${ghprbAuthorRepoGitUrl} branch ${ghprbActualCommit}"
+        ci_git_branch = "${ghprbActualCommit}"
+        ci_git_url = "${ghprbAuthorRepoGitUrl}"
+  } else {
+        echo "Checkout ${ghprbAuthorRepoGitUrl} main"
+        ci_git_branch = 'main'
+        ci_git_url = 'https://github.com/metal3-io/project-infra.git'
+    }
+
+    int rand = new SecureRandom()
+    VM_KEY = (1..4).collect {
+        ('a'..
+      'z').join('')[rand.nextInt(26)]
+  }.join('')
+
+    agent_label="metal3ci-large-${IMAGE_OS}"
+}
+
+pipeline {
+    agent none
+    environment {
+        METAL3_CI_USER = 'metal3ci'
+        REPO_ORG = "${PROJECT_REPO_ORG}"
+        REPO_NAME = "${PROJECT_REPO_NAME}"
+        UPDATED_REPO = "${ghprbAuthorRepoGitUrl}"
+        REPO_BRANCH = "${ghprbTargetBranch}"
+        UPDATED_BRANCH = "${ghprbActualCommit}"
+        BUILD_TAG = "${env.BUILD_TAG}"
+        PR_ID = "${ghprbPullId}"
```

Comment:
where this can be used

Reply:
REPO_BRANCH is the branch where we are running the test(main,1.6 etc) and UPDATED_BRANCH is the commit on which we want to run test. BUILD_TAG is used to name the log folders. PR_ID I am not sure why we need it, I haven't been find where this is being used. I will remove it and see if anything breaks.